adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -86,6 +89,7 @@ private void removeListener(OrderedRealmCollection<T> data) {
      * @return the number of items.
      */
     @Override
+    @CheckResult
```

Comment:
I think all these `@CheckStyle` are not really needed. This annotation should primarily be used on methods where it would be an error to ignore the return value, e.g. because the method has side-effects and return a modified result, e.g. `string.trim()` https://developer.android.com/reference/android/support/annotation/CheckResult.html.

Ignoring the return value of these methods are not technically an error (although it would be dead code).

What do you think @realm/java